frontend/src/pages/submission/[id].jsx

```diff
@@ -0,0 +1,53 @@
+import React, { useEffect, useState } from 'react';
+import { useRouter } from 'next/router';
+import { Paper, Typography, TextField } from '@mui/material';
+
+const submission_service_port = 5433;
+
+export default function SubmissionPage() {
+  const router = useRouter();
+  // Get the dynamic part of the URL
+  const { id } = router.query;
+
+  const [submission, setSubmission] = useState([]);
+
+  useEffect(() => {
+    const fetchSubmissionDetails = async () => {
+      try {
+        console.log("Submission ID: " + id);
+        const response = await fetch(`http://localhost:${submission_service_port}/submission/${id}`);
+        if (!response.ok)
```

Comment:
This part might not be necessary as any non 200 responses are automatically caught in the `catch` block.